File: app/src/main/java/com/example/morro/telecomando/Core/Song.java
package com.example.morro.telecomando.Core;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

/**
 * data model being displayed by the RecyclerView
 */

public class Song {
    private String itemPath;
    private String title;
    private String artist;
    private String album;
    private String year;

    public String getTitle() {
        return title;
    }

    public String getArtist() {
        return artist;
    }

    public String getAlbum() {
        return album;
    }

    public String getYear() {
        return year;
    }

    public String getItemPath() {
        return itemPath;
    }

    public Song(String title, String artist, String album, String year, String itemPath) {
        this.itemPath = itemPath;
        this.title = title;
        this.artist = artist;
        this.album = album;
        this.year = year;
    }

    /*
    public static ArrayList<Item> createTrackList(String content,ArrayList<Item> items) {
        String[] lines = content.split(System.getProperty("line.separator"));
        int nLines = lines.length;
        for(int i=1; i <= nLines ; i++){
            items.add(new Item(lines[0]));
        }

        return items;
    }
    */

    public static ArrayList<Song> createTrackList(int num) {
        ArrayList<Song> songs = new ArrayList<Song>();

        for (int i = 1; i <= num; i++) {
            songs.add(new Song("Title", "Artist", "Album", "Year", "path"));
        }

        return songs;
    }


    public String getJson(){
        JSONObject song = new JSONObject();
        try {
            song.put("title", title);
            song.put("artist", artist);
            song.put("album", album);
            song.put("year", year);
            song.put("path", itemPath);
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return song.toString();
    }

}

File: app/src/main/java/com/example/morro/telecomando/core/Song.kt
package com.example.morro.telecomando.core

import org.json.JSONObject
import java.util.*

/**
 * data model being displayed by the RecyclerView
 */
data class Song(val title: String, val artist: String, val album: String, val year: String, val itemPath: String) {

    fun getJson(): String {
        val song = JSONObject()
        song.put("title", title)
        song.put("artist", artist)
        song.put("album", album)
        song.put("year", year)
        song.put("path", itemPath)
        return song.toString()
    }

    override fun toString(): String {
        return "$title ($artist)"
    }

    /* works like static functions in java classes */
    companion object {
        @JvmStatic
        fun buildDummyTrackList(num: Int = 1): ArrayList<Song> {
            val songs = ArrayList<Song>()
            for (i in 1..num) {
                songs.add(Song("Title$i", "Artist$i", "Album$i", "Year$i", "path$i"))
            }
            return songs
        }
    }
}